### PR TITLE
Proof of concept to use goss for health checks

### PR DESCRIPTION
**Pull request by kasperg**
This is basically the result of following 
https://medium.com/@aelsabbahy/docker-1-12-kubernetes-simplified-health-checks-and-container-ordering-with-goss-fa8debbe676c

The result of running it after starting a container:

``` bash
14:45 $ docker inspect -f '{{ range .State.Health.Log }}{{ println "======\nStart:" .Start }}{{ .Output }}{{end}}' a7a51b6ef6da3ef698f7305d324f1d9f0ad440f129735975031d5da3a88141d4
======
Start: 2018-09-04 12:45:38.2667981 +0000 UTC
FSFFF

Failures/Skipped:

Port: tcp6:9000: listening:
Expected
    <bool>: false
to equal
    <bool>: true
Port: tcp6:9000: ip: skipped

Process: php-fpm7.2: running:
Expected
    <bool>: false
to equal
    <bool>: true

Command: composer: exit-status:
Expected
    <int>: 255
to equal
    <int>: 0

Command: drush: exit-status:
Expected
    <int>: 255
to equal
    <int>: 0

Total Duration: 0.195s
Count: 5, Failed: 4, Skipped: 1
======
Start: 2018-09-04 12:45:39.5403977 +0000 UTC
FFSF.

Failures/Skipped:

Process: php-fpm7.2: running:
Expected
    <bool>: false
to equal
    <bool>: true

Port: tcp6:9000: listening:
Expected
    <bool>: false
to equal
    <bool>: true
Port: tcp6:9000: ip: skipped

Command: composer: exit-status:
Expected
    <int>: 255
to equal
    <int>: 0

Total Duration: 0.203s
Count: 5, Failed: 3, Skipped: 1
======
Start: 2018-09-04 12:45:40.8227662 +0000 UTC
FSF..

Failures/Skipped:

Port: tcp6:9000: listening:
Expected
    <bool>: false
to equal
    <bool>: true
Port: tcp6:9000: ip: skipped

Process: php-fpm7.2: running:
Expected
    <bool>: false
to equal
    <bool>: true

Total Duration: 0.248s
Count: 5, Failed: 2, Skipped: 1
======
Start: 2018-09-04 12:45:42.1592481 +0000 UTC
.....

Total Duration: 0.215s
Count: 5, Failed: 0, Skipped: 0
```

### CODE REVIEW

**Comment by xendk** [comment]
Haven't this been implemented?

File: Dockerfile-5.6 [hunk 1]
```diff
@@ -99,6 +104,6 @@ RUN phpenmod drupal-recommended
 
 ENV PHP_DEFAULT_EXTENSIONS calendar ctype curl dom exif fileinfo ftp gd gettext iconv json mcrypt mysqli mysqlnd opcache pdo pdo_mysql phar posix readline shmop simplexml soap sockets sqlite sysvmsg sysvsem sysvshm tokenizer wddx xml xmlreader xmlwriter xsl mbstring zip
 
-HEALTHCHECK CMD ["sh", "-c", "[ -e /tmp/docker-finished-init ]"]
+HEALTHCHECK --interval=1s --timeout=5s CMD goss -g /goss/goss.yaml validate
```
Comment (xendk) [inline Dockerfile-5.6:107]: To be complete, shouldn't we add a file https://github.com/aelsabbahy/goss/blob/master/docs/manual.md#file test on /tmp/docker-finished-init

**Comment by xendk** [comment]
@kasperg @arnested @danquah Do we think it's worthwhile to get this up to date and merged?

**Comment by kasperg** [comment]
@xendk: I don't know:

1. I like health checks
2. I like Goss but I am currently the only one with experience using the tool. If we merged this more people will be exposed to it.

Whether this is the right thing to do(tm) is a different matter. Having proposed the change myself I do not think I am the right person to decide this or not.

**Comment by xendk** [comment]
I like goss and healthchecks too, but... Is it really worth burning CPU cycles to check every second that it's listening on a port and four different executables runs and returns zero exit code?

**Comment by arnested** [comment]
I, too, like health checks (and doesn't hate goss).

But I'll agree this might seem a little, unnecessary heavy.

The simplest / most light way health check for checking a port is accessible is probably by using `lsof` but unfortunately, it requires additional permissions to be able to run it in Docker. We use it in a few places to do health check for databases, e.g. https://github.com/reload/dj/blob/master/docker-compose.yml#L111-L114.

**Pull request closed without merging**